lib/streamlit/elements/lib/built_in_chart_utils.py

```diff
@@ -165,15 +167,27 @@ def generate_chart(
 
     # At this point, x_column is only None if user did not provide one AND df is empty.
 
+    if not horizontal:
+        x_encoding = _get_x_encoding(
+            df, x_column, x_from_user, x_axis_label, chart_type
+        )
+        y_encoding = _get_y_encoding(df, y_column, y_from_user, y_axis_label)
+    else:
+        # Handle horizontal bar chart - switches x and y encodings:
+        x_encoding = _get_y_encoding(df, y_column, y_from_user, x_axis_label)  # type: ignore[assignment]
+        y_encoding = _get_x_encoding(
+            df, x_column, x_from_user, y_axis_label, chart_type
+        )  # type: ignore[assignment]
```

Comment:
This feels a bit hacky since it will assign `alt.X` to y and `alt.Y` to x. It probably works because they are extending the same base class without too many changes, but it could break if things change in Altair. Would it be possible to just change the parameters? 

```python
 x_encoding = _get_x_encoding(df, y_column, y_from_user, x_axis_label, chart_type) 
```

Reply:
It does seem hacky, but it was the only solution I found to reliably create a horizontal bar chart 😓 
Just changing the param produced the below:
<img width=500 src=https://github.com/streamlit/streamlit/assets/63436329/3f392aa4-f8fd-4c53-8e26-38b4a310f442 />

Reply:
Oh, I believe this might be the reason why it's not working when you just change the parameters:

https://github.com/streamlit/streamlit/blob/a94106dcbe4f667b8bfec0140615bbd1d3938e69/lib/streamlit/elements/lib/built_in_chart_utils.py#L894-L897

It's making the wrong column ordinal

Reply:
One alternative might be to replace `ChartType.BAR` with 

```
    VERTICAL_BAR = {"mark_type": "bar", "command": "bar_chart", "horizontal": False}
    HORIZONTAL_BAR = {"mark_type": "bar", "command": "bar_chart", "horizontal": True}
```

and then do this in `_get_x_encoding_type`:

```
if chart_type == ChartType.VERTICAL_BAR and not _is_date_column(df, x_column): 
     return "ordinal"
```

and this in `_get_y_encoding_type`:

```
if chart_type == ChartType.HORIZONTAL_BAR and not _is_date_column(df, y_column): 
     return "ordinal"
```

And just do the selection of `VERTICAL_BAR` vs `HORIZONTAL_BAR` type in the `bar_chart` command.

Reply:
Really appreciate the feedback! Have incorporated the above and made other small tweaks to get it to render properly with just changing params 👍🏼

Reply:
Awesome :)